src/pricecypher/encoders/JsonEncoder.py

```diff
@@ -0,0 +1,21 @@
+import dataclasses
+import numpy as np
+
+from json import JSONEncoder
+
+
+class PriceCypherJsonEncoder(JSONEncoder):
+    """
+    JSON encoder that can properly serialize dataclasses and numpy numbers.
+    """
+    def default(self, obj):
```

Comment:
Python 3.10 has added a switch with matching, might look better than this.

Reply:
How does that work with almost every current project that uses python 3.9? I really don't feel like setting up polyfills or whatever, if that's even possible. Or even worse bumping the major version and updating almost everything that uses this script just so we can use a switch statement for this one class.